### PR TITLE
Emphasize the meaning of "source $stdenv/setup" and add a hint to pre…

### PR DESCRIPTION
**Pull request by Miko4**
This is not a huge difference, but I think it's clearer now. I did misunderstood it, witch did lead to #11098.

### CODE REVIEW

File: doc/stdenv.xml [hunk 1]
```diff
@@ -100,9 +100,10 @@ where the builder can do anything it wants, but typically starts with
 source $stdenv/setup
 </programlisting>
 
-to let <literal>stdenv</literal> set up the environment (e.g., process
-the <varname>buildInputs</varname>).  If you want, you can still use
-<literal>stdenv</literal>’s generic builder:
+to let <literal>stdenv</literal> set up the environment. If this is
+not the case, the <varname>buildInputs</varname> are not aviable.  If
```
Comment (vizanto) [inline doc/stdenv.xml]: available

**Comment by lethalman** [comment]
I understand your sentiment, however that's also still much naive. `buildInputs` are available, but not in `$PATH`. But then that should expand a whole chapter on how stdenv setup works.

**Comment by joachifm** [comment]
@Miko4 do you intend to continue working on this? Otherwise, I'll close this as abandoned.

**Comment by vcunat** [comment]
Perhaps we should just stress that typically you only want to tweak the generic build by setting some variables?

**Comment by joachifm** [comment]
Closing as abandoned.

**Pull request closed without merging**